src/wasm-binary.h

```diff
@@ -1051,10 +1051,12 @@ class WasmBinaryBuilder {
   const std::vector<char>& input;
   std::istream* sourceMap;
   std::pair<uint32_t, Function::DebugLocation> nextDebugLocation;
+  bool DWARF = false;
 
   size_t pos = 0;
   Index startIndex = -1;
   std::set<Function::DebugLocation> debugLocation;
+  size_t codeSectionLocation;
```

Comment:
This should maybe be an `Address` rather than a `size_t` (or something that depends on the target architecture rather than the host architecture).

Reply:
Hmm, it is the offset in the wasm binary. I felt `size_t` makes sense as it's basically an offset in a native file. And `Address` would depend on whether this is `wasm32` or `wasm64`.

Reply:
Oh I missed this; can we use the offset in the section rather than the offset in the binary? I believe all the debug info uses section offsets rather than file offsets.

Also, per the spec, basically all of the LEB-encoded numbers in the wasm format are limited to 32 bits. So it wouldn't be possible today to even encode a section over 4G.

Reply:
We do use the offset in the section for instructions, but this is how we compute that - we note the section's start here, then subtract it later when we see where an expression is emitted.

About the 32-bit LEBs, in theory each section can be less than 4GB but their total size might be larger, in which case this being a `size_t` would still work :wink: but I don't feel strongly about that corner case if you think it's not important to handle.

Reply:
All the custom sections use section offsets too, so if we use binary offsets for everything we'll need to keep this sectionOffset for every section. So I guess that's just a matter of where you want to put your arithmetic. 
But that can be for a further later if want. For this variable specifically, then maybe let's just use a 64-bit type everywhere (i.e. even where `size_t` is 32 bits). Then it will always be consistent and if we run into size issue on 32-bit platforms, the symptom will be something obvious like allocation failure rather than integer overflow.